RoboJackets/robocup-software
# Shoot only

## Description
This PR outlines the basics of a shoot only offense.  In this offense strategy, one robot (the closest to the ball that is playing offense) will attempt to obtain the ball and shoot the ball at the opponents goal.

EDIT: PR #2069 was merged into this branch. It adds a pivot step before any shot on goal to improve line kick consistency, but only does so at the Position level, which is not ideal (should be a planner, but planners are hard to compose).

## Associated / Resolved Issue
[coordinate shoot-only offense](https://app.clickup.com/t/8677u14px)

## Steps to Test
### Test Case 1
1. make perf && make run-sim
2. assign any number of robots to offense
3. click start (the robots will reassign is switched to offense and closer to the ball while sim is playing if you want to see that just start then assign robots to offense)

**Expected result:**The robot closest to the ball should get the ball and shoot it into the net.  Currently the robot keeps doing this on loop which may be undesired, but I would assume the coach will reassign him so I have the scorer on loop.

## Key Files to Review
Messages
 * rj_msgs/request/ScorerRequest.msg
 * rj_msgs/response/ScorerResponse.msg
 * soccer/src/soccer/strategy/agent/communication.*pp

Offense
 * soccer/src/soccer/strategy/agent/position/offense.*pp

## Review Checklist
